core/src/main/java/com/datastrato/gravitino/authorization/AccessControlManager.java

```diff
@@ -144,4 +160,17 @@ public boolean isServiceAdmin(String user) {
   public boolean isMetalakeAdmin(String user) {
     return adminManager.isMetalakeAdmin(user);
   }
+
+  private void checkMetalakeExists(String metalake) throws NoSuchMetalakeException {
+    try {
+      NameIdentifier metalakeIdent = NameIdentifier.ofMetalake(metalake);
+      if (!store.exists(metalakeIdent, Entity.EntityType.METALAKE)) {
+        LOG.warn("Metalake {} does not exist", metalakeIdent);
+        throw new NoSuchMetalakeException(METALAKE_DOES_NOT_EXIST_MSG, metalakeIdent);
+      }
+    } catch (IOException e) {
+      LOG.error("Failed to do storage operation", e);
+      throw new RuntimeException(e);
+    }
+  }
```

Comment:
I think this part should be put into `UserGroupManager`, and put into a single lock, right?

Reply:
Done.